client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -125,22 +127,42 @@ private synchronized void putBlockId(
 
   public RssShuffleManager(SparkConf conf, boolean isDriver) {
     this.sparkConf = conf;
+
+    // set & check replica config
+    dataReplica = sparkConf.getInt(RssClientConfig.RSS_DATA_REPLICA,
+      RssClientConfig.RSS_DATA_REPLICA_DEFAULT_VALUE);
+    dataReplicaWrite =  sparkConf.getInt(RssClientConfig.RSS_DATA_REPLICA_WRITE,
+      RssClientConfig.RSS_DATA_REPLICA_WRITE_DEFAULT_VALUE);
+    dataReplicaRead =  sparkConf.getInt(RssClientConfig.RSS_DATA_REPLICA_READ,
+      RssClientConfig.RSS_DATA_REPLICA_READ_DEFAULT_VALUE);
+    LOG.info("Create RssShuffleManager with replica config ["
+      + dataReplica + ":" + dataReplicaWrite + ":" + dataReplicaRead + "]");
+    if (dataReplica == 1) {
+      if (dataReplicaWrite != 1 || dataReplicaRead != 1) {
+        throw new RuntimeException("Replica config is invalid, recommend replica.write + replica.read > replica");
+      }
+    } else if (dataReplica > 1) {
+      if (dataReplica <= dataReplicaWrite + dataReplicaRead) {
+        throw new RuntimeException("Replica config is unsafe, recommend replica.write + replica.read > replica");
+      }
+    }
```

Comment:
Could you put them in a function and make it unit testable.

Reply:
ok

Reply:
Add a QuorumConfigTest